ci/docker/python-wheel-manylinux-x-y.dockerfile

```diff
@@ -0,0 +1,94 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+ARG base
+FROM ${base}
+
+ARG arch
+ARG arch_short
+ARG manylinux
+
+ENV MANYLINUX_VERSION=${manylinux}
+
+# Install basic dependencies
+RUN yum install -y git flex curl autoconf zip perl-IPC-Cmd wget
+
+ENV CPYTHON_VERSION=cp38
+ENV PATH=/opt/python/${CPYTHON_VERSION}-${CPYTHON_VERSION}/bin:${PATH}
```

Comment:
It seems that we can unify `ci/docker/python-wheel-manylinux-201x.dockerfile` and this file without this.
If we remove this, what error is happen?

Reply:
Manylinux2014 container (CENTOS 7) has python 2.7.5 installed by default, but manylinux_2_28 (Almalinux 8) does not:

```bash
$ docker run -it --rm quay.io/pypa/manylinux2014_x86_64 -- python --version
Python 2.7.5
$ docker run -it --rm quay.io/pypa/manylinux_2_28_x86_64 -- python --version
/usr/local/bin/manylinux-entrypoint: line 8: exec: python: not found
```

For this reason it might be better to keep the following in the unified container as Python 3.8 on the PATH should override Python 2.7.5

```Dockerfile
ENV CPYTHON_VERSION=cp38
ENV PATH=/opt/python/${CPYTHON_VERSION}-${CPYTHON_VERSION}/bin:${PATH}
```

Something else to consider: manylinux_2014 (CENTOS 7) and manlinux_2_28 (AlmaLinux 8) are slightly different operating systems, although they are both based on RHEL. Therefore, there may be slight differences needed for each OS in future which might require separate Dockerfiles. This is a guess on my part as I'm not very experienced with Redhat system.

Summarised: there may be some small risk of OS incompatibility for a unified Dockerfile vs reducing the cost of maintaining multiple Dockerfiles.

If this is OK, I will make a unified Dockerfile during the week.

Reply:
For example: https://www.liquidweb.com/kb/centos-vs-almalinux/

Reply:
Do we need to use Python in `docker build`?
If we don't need to use Python in `docker build`, we don't need `python` in `$PATH`.

Reply:
> Do we need to use Python in `docker build`? If we don't need to use Python in `docker build`, we don't need `python` in `$PATH`.

It is used by ninja and vcpkg.

Reply:
I see. Then we need to keep this.

Unified Dockerfile can have this unconditionally. Because this is no-op with manylinux2014 container.

Reply:
Unified Dockerfile created